editor/source/editor/field.js

```diff
@@ -215,16 +215,24 @@ export class PartialsField extends ListField {
   }
 
   renderCollapsedPartial(editor, partialItem) {
-    return html`
-      <div class="selective__list__item__drag"><i class="material-icons">drag_indicator</i></div>
-      <div class="selective__list__item__label" data-index=${partialItem['index']} @click=${this.handleItemExpand.bind(this)}>
-        ${partialItem['partialConfig']['label']}
-      </div>
-      <div class="selective__list__item__preview" data-index=${partialItem['index']} @click=${this.handleItemExpand.bind(this)}>
-        ${partialItem['partialConfig']['preview_field']
-          ? autoDeepObject(this.value[partialItem['index']]).get(partialItem['partialConfig']['preview_field'])
-          : ''}
-      </div>`
+    let previewField = partialItem['partialConfig']['preview_field'];
```

Comment:
This seems kind of kludgy ... is there a better way to do it? I basically want to make the whole "preview" an optional block depending on whether there's a deepObject...

Reply:
The `ListField` calls the `renderPreview` which does the logic for what to show as the preview.

https://github.com/grow/selective-edit/blob/master/js/selective/field.js#L508

You could do something similar to make it only return the preview div if there is a value instead of having to duplicate everything.

Reply:
Yeah this is better. When you pull everything into JS functions it makes more sense.